internal/core/items/text.rs

```diff
@@ -674,33 +674,52 @@ pub struct TextInputVisualRepresentation {
     pub selection_range: core::ops::Range<usize>,
     /// The position where to draw the cursor, as byte offset within the text.
     pub cursor_position: Option<usize>,
+    text_without_password: Option<String>,
+    password_character: char,
 }
 
 impl TextInputVisualRepresentation {
     /// If the given `TextInput` renders a password, then all characters in this `TextInputVisualRepresentation` are replaced
     /// with the given password character and the selection/preedit-ranges/cursor position are adjusted.
```

Comment:
```suggestion
    /// If the given `TextInput` renders a password, then all characters in this `TextInputVisualRepresentation` are replaced
    /// with the password character and the selection/preedit-ranges/cursor position are adjusted.
    /// If `password_character_fn`, it is called lazily to query the password character, otherwise a default is used.
```